src/base/dropdown-element.ts

```diff
@@ -24,43 +24,71 @@ export type DropdownButtonVariant =
   | "dark";
 export type DropDirection = "left" | "right" | "up" | "down";
 
+/**
+ * @event sgds-show - Emitted event when show instance is called
+ * @event sgds-shown - Emitted event when dropdown has been made visible to the user and CSS transitions have completed
+ * @event sgds-hide - Emitted event when hide instance is called
+ * @event sgds-hidden - Emitted event when dropdown has hidden to the user and CSS transitions have completed
+ */
+
 export class DropdownElement extends SgdsElement {
   static styles = SgdsElement.styles;
 
+  /** @internal */
   myDropdown: Ref<HTMLElement> = createRef();
+  /** @internal */
   bsDropdown: Dropdown = null;
 
+  /** Controls auto-flipping of menu */
   @property({ type: Boolean, reflect: true })
   noFlip = false;
+
+  /** When true, aligns right edge of menu with right edge of button */
   @property({ type: Boolean, reflect: true })
   menuAlignRight = false;
+
+  /** The drop position of menu relative to the toggle button */
   @property({ type: String, reflect: true })
   drop: DropDirection = "down";
+
+  /** Additional configuration to pass to Popper.js. See https://popper.js.org/ for config opts */
   @property({ type: Object })
   popperOpts = {};
+
+  /** Forwards value to id attribute of toggle button of Dropdown. An unique id generated by default */
   @property({ type: String })
   togglerId = genId("dropdown", "button");
```

Comment:
can help me add typing  "string" here. I think the custom element analzyer cannot detect genId returns a string 

`togglerId : string = genId("dropdown", "button")`

Reply:
Resolved in [40d2e42](https://github.com/GovTechSG/sgds-web-component/pull/54/commits/40d2e42a0eb72c750567fb021865f04ca94c517c)